backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/backup/StartVmBackupCommand.java

```diff
@@ -553,6 +557,40 @@ protected Set<Guid> getFromCheckpointDisksIds(Guid fromCheckpointId) {
         return fromCheckpointDisksIds;
     }
 
+    protected boolean validateCheckpoint(Guid checkpointId) {
+        List<DiskImage> images = vmCheckpointDao.getDisksByCheckpointId(checkpointId);
+        for (DiskImage image : images) {
+            VdsmImageLocationInfo locationInfo = new VdsmImageLocationInfo(
+                        image.getStorageIds().get(0),
+                        image.getId(),
+                        image.getImageId(),
+                        null);
+
+            VolumeBitmapCommandParameters parameters =
+            new VolumeBitmapCommandParameters(
+                    getStoragePoolId(),
+                    locationInfo,
+                    checkpointId.toString());
+            parameters.setVdsId(getVdsId());
+            parameters.setEndProcedure(ActionParametersBase.EndProcedure.COMMAND_MANAGED);
+            parameters.setParentCommand(getActionType());
+            parameters.setParentParameters(getParameters());
+
+            ActionReturnValue returnValue = runInternalActionWithTasksContext(ActionType.CheckVolumeBitmap, parameters);
```

Comment:
I haven't had  thorough look yet, and the general idea sounds good to me, but I'm not sure about having such side effects (checkpoint removal) in the validation phase

Reply:
@bennyz : Thanks for checking already.

Still have to validate the checkpoints before starting a NBDServer.
Would it also be a good idea to for example just redefine the checkpoints when starting a VM?

Then we need 2 places to check the existing DB checkpoints.
- When starting the VM (we could just redefine them, current code already does a cleanup of the checkpoints if redefine fails)
- When starting a NBDServer for an Down VM, we should use the newly introduced 'CheckVolumeBitmapCommand' to validate the checkpoint passed.